ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1097,6 +1100,14 @@ func (r *RayClusterReconciler) buildHeadPod(ctx context.Context, instance rayv1.
 	if len(r.headSidecarContainers) > 0 {
 		podConf.Spec.Containers = append(podConf.Spec.Containers, r.headSidecarContainers...)
 	}
+	rayContainer := podConf.Spec.Containers[utils.RayContainerIndex]
```

Comment:
can you create a utility function `InjectPreStopHookIfNeeded()` and `createPreStopLifecycle` is no longer needed.

```go
func InjectPreStopHookIfNeeded(rayContainer) {
  ...
}
```